### PR TITLE
update / fix development guide

### LINKED ISSUE #756: dev: could not apply CRDs in development docs
**Describe the bug**
Instructions in [development.md](https://github.com/kcp-dev/kcp/blob/main/DEVELOPMENT.md) are potentially  not up to date as some steps don't work.

**To Reproduce**
1. clone from ea0953d357b02ea69dba2322c5fbf4170492c73b
2. run `go run ./cmd/kcp start` keep running
3. api-resources
    
    ```
    k api-resources
    NAME                          SHORTNAMES   APIVERSION                             NAMESPACED   KIND
    configmaps                    cm           v1                                     true         ConfigMap
    events                        ev           v1                                     true         Event
    limitranges                   limits       v1                                     true         LimitRange
    namespaces                    ns           v1                                     false        Namespace
    resourcequotas                quota        v1                                     true         ResourceQuota
    secrets                                    v1                                     true         Secret
    serviceaccounts               sa           v1                                     true         ServiceAccount
    customresourcedefinitions     crd,crds     apiextensions.k8s.io/v1                false        CustomResourceDefinition
    tokenreviews                               authentication.k8s.io/v1               false        TokenReview
    localsubjectaccessreviews                  authorization.k8s.io/v1                true         LocalSubjectAccessReview
    selfsubjectaccessreviews                   authorization.k8s.io/v1                false        SelfSubjectAccessReview
    selfsubjectrulesreviews                    authorization.k8s.io/v1                false        SelfSubjectRulesReview
    subjectaccessreviews                       authorization.k8s.io/v1                false        SubjectAccessReview
    certificatesigningrequests    csr          certificates.k8s.io/v1                 false        CertificateSigningRequest
    leases                                     coordination.k8s.io/v1                 true         Lease
    events                        ev           events.k8s.io/v1                       true         Event
    flowschemas                                flowcontrol.apiserver.k8s.io/v1beta1   false        FlowSchema
    prioritylevelconfigurations                flowcontrol.apiserver.k8s.io/v1beta1   false        PriorityLevelConfiguration
    clusterrolebindings                        rbac.authorization.k8s.io/v1           false        ClusterRoleBinding
    clusterroles                               rbac.authorization.k8s.io/v1           false        ClusterRole
    rolebindings                               rbac.authorization.k8s.io/v1           true         RoleBinding
    roles                                      rbac.authorization.k8s.io/v1           true         Role
    clusterworkspaces                          tenancy.kcp.dev/v1alpha1               false        ClusterWorkspace
    clusterworkspacetypes                      tenancy.kcp.dev/v1alpha1               false        ClusterWorkspaceType
    ```

4. `kubectl apply -f config/` with kubectl 1.21, and it doesn't work, error `error: error reading [config/]: recognized file extensions are [.json .yaml .yml]`

5. try with `-R` also fails:
      
      ```
      kubectl apply -R -f config/
      
      customresourcedefinition.apiextensions.k8s.io/apiresourceimports.apiresource.kcp.dev created
      customresourcedefinition.apiextensions.k8s.io/negotiatedapiresources.apiresource.kcp.dev created
      customresourcedefinition.apiextensions.k8s.io/apibindings.apis.kcp.dev created
      customresourcedefinition.apiextensions.k8s.io/apiexports.apis.kcp.dev created
      customresourcedefinition.apiextensions.k8s.io/apiresourceschemas.apis.kcp.dev created
      Warning: resource customresourcedefinitions/clusterworkspaces.tenancy.kcp.dev is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. kubectl apply should only be used on resources created declaratively by either kubectl create --save-config or kubectl apply. The missing annotation will be patched automatically.
      customresourcedefinition.apiextensions.k8s.io/clusterworkspaces.tenancy.kcp.dev configured
      Warning: resource customresourcedefinitions/clusterworkspacetypes.tenancy.kcp.dev is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. kubectl apply should only be used on resources created declaratively by either kubectl create --save-config or kubectl apply. The missing annotation will be patched automatically.
      customresourcedefinition.apiextensions.k8s.io/clusterworkspacetypes.tenancy.kcp.dev configured
      customresourcedefinition.apiextensions.k8s.io/workspaces.tenancy.kcp.dev created
      customresourcedefinition.apiextensions.k8s.io/workspaceshards.tenancy.kcp.dev created
      customresourcedefinition.apiextensions.k8s.io/workloadclusters.workload.kcp.dev created
      Warning: resource clusterworkspacetypes/universal is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. kubectl apply should only be used on resources created declaratively by either kubectl create --save-config or kubectl apply. The missing annotation will be patched automatically.
      clusterworkspacetype.tenancy.kcp.dev/universal configured
      Warning: resource clusterroles/system:kcp:universal-clusterworkspacetype-use is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. kubectl apply should only be used on resources created declaratively by either kubectl create --save-config or kubectl apply. The missing annotation will be patched automatically.
      clusterrole.rbac.authorization.k8s.io/system:kcp:universal-clusterworkspacetype-use configured
      Warning: resource clusterrolebindings/system:kcp:authenticated:universal-clusterworkspacetype-use is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. kubectl apply should only be used on resources created declaratively by either kubectl create --save-config or kubectl apply. The missing annotation will be patched automatically.
      clusterrolebinding.rbac.authorization.k8s.io/system:kcp:authenticated:universal-clusterworkspacetype-use configured
      namespace/default created
      namespace/default unchanged
      unable to recognize "config/root/workspaceshard.yaml": no matches for kind "WorkspaceShard" in version "tenancy.kcp.dev/v1alpha1"
      Error from server (Forbidden): error when creating "config/root/clusterworkspace-default.yaml": clusterworkspaces.tenancy.kcp.dev "default" is forbidden: spec.type "Organization" does not exist
      Error from server: error when creating "config/root/clusterworkspacetype-organization.yaml": organization type can only be created in root workspace
      Error from server (BadRequest): error when creating "config/root/secret-shard.yaml": Secret in version "v1" cannot be handled as a Secret: illegal base64 data at input byte 5
      ```

**Expected behavior**
instructions  should work

**Additional context**
kubectl 1.21.0, occurs at commit ea0953d357b02ea69dba2322c5fbf4170492c73b

### PR DESCRIPTION
**Pull request by xrstf**
## Summary
This PR performs minor maintenance on the DEVELOPMENT.md. In #605 the `Cluster` type was renamed to `WorkloadCluster`, but the dev docs were not updated.

I also chose to correct (?) the spelling of "kind", which on their website never refer to themselves as "KinD".

## Related issue(s)
Fixes #756

### CODE REVIEW

**Comment by ncdc** [comment]
Hi @xrstf, thanks for your first-time contribution to kcp!!!

We've made a ton of changes since this doc was last updated. Would you mind doing some revisions to your PR? If so, this is what I would recommend:

1. Remove the entire "Build and run Cluster Controller" section
2. https://github.com/kcp-dev/kcp/blob/main/docs/syncer.md now more thoroughly describes how to register a workload cluster. I would remove everything under the "Test the registration of a Physical Cluster" section and replace it with a link to this syncer.md doc.
3. Remove the last sentence of the doc (about kind and `Launch cluster controller`)

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -14,12 +14,16 @@ kubectl api-resources
 # Build and run Cluster Controller
```
Comment (sttts) [inline DEVELOPMENT.md:14]: actually I think the whole section should go. cluster-controller is still there, but totally untested and kcp itself has the same controller integrated. So nobody has to run it in its own process. 

Also installing the CRDs via `kubectl apply` is not needed anymore.

Comment (sttts) [reply DEVELOPMENT.md:14]: Finally `# Test the registration of a Physical Cluster` is also a relict of the past. Push mode is gone in main branch, and there is a cli plugin (`kubectl workloads sync`) to setup a syncer.

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -44,6 +48,7 @@ sed -e 's/^/    /' ${HOME}/.kube/config | cat contrib/examples/cluster.yaml - |
 ```
 
 # Using `kcp` as a library
+
```
Comment (sttts) [inline DEVELOPMENT.md:51]: If I am not mistaken,  `DefaultConfig` is no more. So the code below must look completely differently now, using option structs.

**Comment by sttts** [comment]
As conclusion of the comments above, I think we should rather delete the whole file as it is totally outdated.

**Comment by xrstf** [comment]
I restructured the docs a little bit.

File: README.md [hunk 3]
```diff
@@ -147,8 +147,25 @@ However, `kcp` is not an acronym.
...
 ## How do I get started?
 
-Check out our [Contributing](CONTRIBUTING.md) and [Development](DEVELOPMENT.md) guides.
+After cloning the repository, you can start kcp on your machine using this command:
 
+```
+go run ./cmd/kcp start
+```
+
+This will build and run your kcp server, and generate a kubeconfig in `.kcp/admin.kubeconfig` you can use to connect to it:
```
Comment (sttts) [inline README.md:156]: :+1:

**Comment by sttts** [comment]
Looks great. Can you squash?

**Comment by sttts** [comment]
Oh, github can do the squashing for us.

**Review by sttts — APPROVED** [review]

**Comment by xrstf** [comment]
Sure, I can squash.

**Pull request merged**